nautobot/utilities/filters.py

```diff
@@ -171,10 +171,26 @@ def filter(self, qs, value):
 
         e.g. `['dcim.device', 'dcim.rack']`
         """
-        # Recursively call `ContentTypeFilter.filter()` for each item in the
-        # list of incoming filter values.
+        if not self.conjoined:
+            q = models.Q()
+
         for v in value:
-            qs = ContentTypeFilter.filter(self, qs, v)
+            if self.conjoined:
+                qs = ContentTypeFilter.filter(self, qs, v)
+            else:
+                try:
+                    app_label, model = v.lower().split(".")
+                except ValueError:
+                    continue
+                q |= models.Q(
+                    **{
+                        f"{self.field_name}__app_label": app_label,
+                        f"{self.field_name}__model": model,
+                    }
+                )
```

Comment:
While I can understand what is going on here, it also feels a tad arcane. Can you please add a little commenting for posterity? Thanks!